templates/search/search_clients_last_seen_v1/query.sql

```diff
@@ -0,0 +1,163 @@
+WITH
+  _derived_search_cols AS (
+    SELECT
+      CASE
+          WHEN engine IS NULL THEN 'none'
+          WHEN STARTS_WITH(engine, 'google') THEN 'google'
+          WHEN STARTS_WITH(engine, 'ddg') OR STARTS_WITH(engine, 'duckduckgo') THEN 'ddg'
+          WHEN STARTS_WITH(engine, 'bing') THEN 'bing'
+          WHEN STARTS_WITH(engine, 'yandex') THEN 'yandex'
+          ELSE 'other'
+      END AS short_engine,
+      COALESCE(organic, 0) + COALESCE(sap, 0) + COALESCE(unknown, 0) + COALESCE(tagged_sap, 0) + COALESCE(tagged_follow_on, 0) AS total_searches,
+      COALESCE(tagged_sap, 0) + COALESCE(tagged_follow_on, 0) AS tagged_searches,
+      COALESCE(ad_click, 0) AS ad_click,
+      COALESCE(search_with_ads, 0) AS search_with_ads,
+      * EXCEPT (ad_click, search_with_ads)
+    FROM
+      search_clients_daily_v7
+    WHERE
+      submission_date_s3 = @submission_date
+  ),
+
+  _derived_engine_searches AS (
+    SELECT
+      STRUCT(
+        short_engine AS key,
+        STRUCT(
+          total_searches,
```

Comment:
For each `short_engine` (so bing, google, ddg, yandex) we will get a monthly breakdown of total_searches, tagged_searches, ad_clicks, and searches_with_ads. We can consider adding other breakdowns here if they could perhaps correlate more closely with revenue.